client/app/account/billing/autoRenew/billing-autoRenew.controller.js

```diff
@@ -725,58 +740,37 @@ angular.module('Billing.controllers').controller('Billing.controllers.AutoRenew'
 
       $scope.$broadcast('paginationServerSide.loadPage', '1', 'serviceTable');
 
-      const userPromise = User.getUser().then((user) => {
-        $scope.user = user;
-      });
-
-      const paymentMeansPromise = $scope.automaticRenewV2Mean.available
-        ? $scope.automaticRenewV2Mean.getPaymentMeans()
-        : $q.when();
-
-      const renewAlignUrlPromise = User.getUrlOf('renewAlign').then((urlRenew) => {
-        $scope.urls.renewAlign = urlRenew;
-      });
-
-      const userGuidePromise = User.getUrlOf('guides').then((guides) => {
-        if (guides && guides.autoRenew) {
-          $scope.guide = guides.autoRenew;
-        }
-      });
-
-      const serviceTypePromise = AutoRenew.getServicesTypes().then((result) => {
-        $scope.servicesTypes = _.map(result, value => ({
-          key: value,
-          text: $translate.instant(`autorenew_service_type_${value}`),
-        }));
-
-        $scope.servicesTypes = $scope.servicesTypes
-          .sort((a, b) => a.text.toLowerCase().localeCompare(b.text.toLowerCase()));
-        $scope.servicesTypes.unshift(ALL_SERVICE_TYPES);
-
-        // sets the model of the type select input.
-        $scope.serviceTypeObject.value = _.find($scope.servicesTypes, {
-          key: $scope.services.selectedType,
-        });
-
-        if (!$scope.serviceTypeObject.value) {
-          $scope.serviceTypeObject.value = ALL_SERVICE_TYPES;
-        }
-      });
+      return $q
+        .all({
+          user: User.getUser(),
+          hasDefaultValidPaymentMean: PaymentMethodHelper.hasDefaultPaymentMethod(),
+          renewAlignUrl: User.getUrlOf('renewAlign'),
+          userGuide: User.getUrlOf('guides'),
+          serviceTypes: AutoRenew.getServicesTypes(),
+          userCertificates: AutoRenew.getUserCertificates(),
+        })
+        .then(({
+          user,
+          hasDefaultValidPaymentMean,
+          renewAlignUrl,
+          userGuide,
+          serviceTypes,
+          userCertificates,
+        }) => {
+          $scope.user = user;
+          $scope.urls.renewAlign = renewAlignUrl;
+          $scope.canDisableAllDomains = userCertificates.includes('domains-batch-autorenew');
```

Comment:
Does it make sense to turn this into an `angular.constant`? (`'domains-batch-autorenew'`)